src/main/java/org/example/AccountService.java

```diff
@@ -9,78 +9,77 @@ public class AccountService {
     Scanner sc = new Scanner(System.in);
     public Account addAccount(){
         Account account = new Account();
-        int type = getAccountType(account);
-        setCurrency(account);
-        setBalance(account,type);
-        System.out.print("Password: ");
-        String password = sc.next();
+        AccountInputService accountInputService = new AccountInputService();
+        AccountType accountType = accountInputService.getAccountType();
+        CurrencyType currencyType = accountInputService.getCurrency();
+        String password = accountInputService.accountConfirmPassword();
+        float balance = accountInputService.getBalance(accountType);
+        account.setBalance(balance);
+        account.setType(accountType);
+        account.setCurrency(currencyType);
         account.setOpen(true);
         account.setPassword(password);
         return account;
     }
-    public Object loginAccount(Session session,int accountId, String accountPassword){
+    public void setAccountInfo(Session session, Customer customer, Account account){
+        AccountService accountService = new AccountService();
+        Transaction tx = session.beginTransaction();
+        account.setCustomer(customer);
+        session.save(account);
+        System.out.println("Account Opened Successfully!");
+        tx.commit();
+    }
+    public Account loginAccount(Session session){
+        AccountInputService accountInputService = new AccountInputService();
+        System.out.print("Enter account ID: ");
+        int accountId = sc.nextInt();
+        String accountPassword = accountInputService.getAccountPassword();
         String query = "From Account where id=:id and password=:password and isOpen=true";
         Query q = session.createQuery(query);
         q.setParameter("id",accountId);
         q.setParameter("password",accountPassword);
-        Object result = q.uniqueResult();
-        if(result==null){
+        Object accountResult = q.uniqueResult();
+        if(accountResult==null){
             System.out.println("invalid input");
             System.out.println("Do you want again input password: y/n ");
             char latter  = sc.next().charAt(0);
             if(latter=='y') {
-                return loginAccount(session, accountId, accountPassword);
+                 return loginAccount(session);
             }else {
                 System.exit(0);
             }
         }
-        return result;
+        Account account = (Account) accountResult;
+        return account;
```

Comment:
you can directly return like

return (Account) accountResult;
why would you need to have another variable and consume memory